### PR TITLE
Make the document addition done log to appear once indexing is over

### LINKED ISSUE #2703: The "document addition done" should appear once indexing is over
**Describe the bug**
Logs appear before the indexation is over

**To Reproduce**
- Run Meilisearch on `main`
- Add documents (a complete dataset, not only 10 or 100 documents)
- Check the logs; you will see the logs saying "indexation is done" arrives immediately after the beginning of the indexing despite the fact that indexing is not over. 
```
[2022-08-24T11:08:24Z INFO  actix_web::middleware::logger] 127.0.0.1 "POST /indexes/movies/documents HTTP/1.1" 202 130 "-" "mieli/0.25.1" 0.257046
[2022-08-24T11:08:24Z INFO  meilisearch_lib::index::updates] document addition done: DocumentAdditionResult { indexed_documents: 31968, number_of_documents: 31968 }
```

**Expected behavior**
The "document addition done" should appear once the indexation is over.

**Meilisearch version:**
`main` branch (between v0.28.1 and v0.29.0, during v0.29.0 sprint)

**Additional context**
Tested locally on my MacOs

### PR DESCRIPTION
**Pull request by evpeople**
The "document addition done"  appear once indexing is over now

# Pull Request

## What does this PR do?
Fixes #2703 

## PR checklist
Please check if your PR fulfills the following requirements:
- [x] Does this PR fix an existing issue?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

File: meilisearch-lib/src/index/updates.rs [hunk 1]
```diff
@@ -348,6 +347,7 @@ impl Index {
         if results.iter().any(Result::is_ok) {
             let _addition = builder.execute()?;
             txn.commit()?;
+            info!("document addition done: {:?}", _addition);
```
Comment (Kerollmops) [inline meilisearch-lib/src/index/updates.rs:350]:
```suggestion
            let addition = builder.execute()?;
            txn.commit()?;
            info!("document addition done: {:?}", addition);
```

File: meilisearch-lib/src/index/updates.rs [hunk 2]
```diff
@@ -336,7 +336,6 @@ impl Index {
                         indexed_documents: count,
                         number_of_documents: count,
                     };
-                    info!("document addition done: {:?}", addition);
                     Ok(addition)
```
Comment (Kerollmops) [inline meilisearch-lib/src/index/updates.rs:339]: Would it be possible to also remove the intermediate `addition` variable here, please?
```rust
Ok(DocumentAdditionResult {
    indexed_documents: count,
    number_of_documents: count,
})
```

**Review by Kerollmops — CHANGES_REQUESTED** [review]
Thank you very much for you PR, just a few little changes to de before I can merge it!
Please, make sure that you do a `cargo fmt` before pushing your commits, just in case 😃

File: meilisearch-lib/src/index/updates.rs (hunk 2 above)
Comment (evpeople) [reply meilisearch-lib/src/index/updates.rs]: Thanks for  your suggestion! I am new to Rust and this is the first time I realize removing the intermediate variable will help

**Review by Kerollmops — APPROVED** [review]
Pretty good job, thank you for your work!
bors merge

**Comment by Kerollmops** [comment]
Strange issue with the lack of the right GLIBC version on the machine, retrying.
bors merge

**Comment by curquiza** [comment]
`Run tests in debug` is the only one to fail, the others are ok, see https://github.com/meilisearch/meilisearch/runs/8094399599?check_suite_focus=true

It means only `cargo --locked` is failing
https://github.com/meilisearch/meilisearch/blob/3efd06168f5ab3f5c11bac942bc9d8605abe8efd/.github/workflows/rust.yml#L41-L57

**Comment by Kerollmops** [comment]
If this is the case @evpeople, could it be possible to do a `cargo update` in a separate commit, please?

**Review by Kerollmops — APPROVED** [review]
bors merge

**Comment by evpeople** [comment]
Sadly, it seems did not work.

**Comment by evpeople** [comment]
I have tried  `cargo test --locked ` on my own Arch Linux, it passed with no error.

**Comment by evpeople** [comment]
The verison of libc on my PC is  2.36

**Comment by curquiza** [comment]
I really don't know
- why the "classic" GtiHub actions (visible at the bottom of the PR) that pass on the PR are working, even for `Run tests in debug`.
- why the GitHub Actions on staging (so the combination of `release-v0.29.0` and `fixBug` branch) does not pass for `Run tests in debug` but works for every other test including `Tests on ubuntu-18.04`, see https://github.com/meilisearch/meilisearch/actions/runs/2961668370

Maybe GitHub is bugging because I change the originated branch from `main` to `release-v0.29.0`, but I 5% believe in this theory...

**Comment by evpeople** [comment]
Should I open another PR to merge my branch(`fixbug`) into `release-v0.29.0` ?

**Comment by curquiza** [comment]
You can try if you have the time, thank you, but to be honest, I'm not confident in this solution 😅

**Comment by evpeople** [comment]
It seems not helpful #2724  
the bors maybe have cached the commit

**Comment by evpeople** [comment]
I can't see more detail https://app.bors.tech/repositories/26457/log#batch-246718 here because of permission denied

**Comment by curquiza** [comment]
It's not related to bors, bors only run the GitHub actions, it's the github actions that fail make bors refuse the merge

**Comment by curquiza** [comment]
waaaah, I don't understand how, but bors merged bot of the PRs!!!

**Comment by curquiza** [comment]
Thanks for your involvement and patience @evpeople

**Pull request merged**